platforms/cosmwasm/src/address.ts

```diff
@@ -113,7 +114,10 @@ export class CosmwasmAddress implements Address {
   // The denomType is "native", "ibc", or "factory"
   private readonly denomType?: string;
 
-  constructor(address: string | Uint8Array | UniversalAddress) {
+  constructor(address: AnyCosmwasmAddress) {
+    if (address instanceof CosmwasmAddress) {
```

Comment:
I'm not positive this `instanceof` check will cause issues but I'm sure its v picky about the check it does. 

If you somehow have a `CosmwasmAddress` from a diff _version_ of the same package it'll fail.  (see https://github.com/algorand/js-algorand-sdk/issues/740 for example)

Mostly pointing it out so if/when we get a weird ticket, this comes to mind easily

Reply:
What about adding a platform value on the address and check that? `if (address.platform === CosmwasmAddress.platform)`